firebase/functions/src/post-lab-write.spec.ts

```diff
@@ -0,0 +1,102 @@
+import 'jest';
+import * as sinon from 'sinon';
+import * as admin from 'firebase-admin';
+import * as setupOfflineTests from 'firebase-functions-test';
+
+const test = setupOfflineTests();
+
+test.mockConfig({
+  firebase: {},
+  fb_service_account: {}
+});
+
+describe('Firebase Functions', () => {
+  let postLabWriteFn, hashDirectoryFn, adminInitStub, oldDatabase;
+
+  describe('postLabWrite', () => {
+    beforeEach(() => {
+      adminInitStub = sinon.stub(admin, 'initializeApp');
```

Comment:
You can use the following construct

```ts
const sandbox = sinon.sandbox.create();

beforeEach(() => {
    adminInitStub = sandbox.stub(admin, 'initializeApp');
});

afterEach(() => {
    sandbox.restore();
});
```

This way, you can just keep on adding stubs to the sandbox and with just one call, you restore everything again.

Reply:
Ah cool! Didn't know that!